moveit_ros/visualization/rviz_plugin_render_tools/src/trajectory_visualization.cpp

```diff
@@ -317,10 +346,20 @@ void TrajectoryVisualization::update(float wall_dt, float ros_dt)
       animating_path_ = true;
       displaying_trajectory_message_ = trajectory_message_to_display_;
       changedShowTrail();
+      trajectory_slider_panel_->update(trajectory_message_to_display_->getWayPointCount());
     }
-    else if (loop_display_property_->getBool() && displaying_trajectory_message_)
-    {  // do loop? -> start over too
-      animating_path_ = true;
+    else if (displaying_trajectory_message_)
+    {
+      if (trajectory_slider_panel_->isVisible() && !loop_display_property_->getBool() &&
```

Comment:
I feel that this **if** statement could be untangled a bit more for the sake of clarity.  Both conditions check for ``trajectory_slider_panel_->isVisible()``, perhaps that could be moved into its own **if** statement.